src/nvim/main.c

```diff
@@ -738,22 +738,17 @@ static void init_locale(void)
   setlocale(LC_NUMERIC, "C");
 # endif
 
-# ifdef LOCALE_INSTALL_DIR    // gnu/linux standard: $prefix/share/locale
-  bindtextdomain(PROJECT_NAME, LOCALE_INSTALL_DIR);
-# else                        // old vim style: $runtime/lang
-  {
-    char_u  *p;
-
-    // expand_env() doesn't work yet, because g_chartab[] is not
-    // initialized yet, call vim_getenv() directly
-    p = (char_u *)vim_getenv("VIMRUNTIME");
-    if (p != NULL && *p != NUL) {
-      vim_snprintf((char *)NameBuff, MAXPATHL, "%s/lang", p);
-      bindtextdomain(PROJECT_NAME, (char *)NameBuff);
-    }
-    xfree(p);
+  char localepath[MAXPATHL] = { 0 };
+  char *exepath = localepath;
+  size_t exepathlen = MAXPATHL;
+  if (os_exepath(exepath, &exepathlen) != 0) {
+    path_guess_exepath(argv0 ? argv0 : "nvim", exepath, sizeof(exepath));
   }
-# endif
+  char *tail = (char *)path_tail_with_sep((char_u *)exepath);
+  *tail = NUL;
+  tail = (char *)path_tail((char_u *)exepath);
+  xstrlcpy(tail, "share/locale", sizeof(localepath) - (tail - localepath));
+  bindtextdomain(PROJECT_NAME, localepath);
```

Comment:
does not regress 1a15cf84c29c984d600f438122edc70be2ac91a1 as long as we assume `@CMAKE_INSTALL_FULL_LOCALEDIR@` would always be in this form.

I wonder if this conforms with BSD conventions.